tools/esdlcmd/README.md

```diff
@@ -0,0 +1,506 @@
+<es:BackendRequest name="request-prep" target="soap:Body/{$query}" xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/" xmlns:es="urn:hpcc:esdl:script">
+    <es:set-value target="RequestValue" value="'foobar'"/>
+</es:BackendRequest>
+```
+
+**WsFoobar-logging-prep.xml**
+
+```xml
+<es:PreLogging name="logging-prep" target="soap:Body/{$query}" xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/" xmlns:es="urn:hpcc:esdl:script">
+    <es:set-value target="LogValue" value="23"/>
+</es:PreLogging>
+```
+
+**FoobarSearch-scripts.xml**
+
+```xml
+<Scripts>
+    <es:BackendRequest name="search-request-prep" target="soap:Body/{$query}" xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/" xmlns:es="urn:hpcc:esdl:script">
+        <es:if test="RequestOption>1">
+            <es:set-value target="HiddenOption" value="true()"/>
+        </es:if>
+    </es:BackendRequest>
+
+    <es:PreLogging name="search-logging-prep" target="soap:Body/{$query}" xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/" xmlns:es="urn:hpcc:esdl:script">
+        <es:if test="RequestOption=1">
+            <es:set-value target="ProductPrice" value="10"/>
+        </es:if>
+    </es:PreLogging>
+</Scripts>
+```
+
+**log-prep.xslt**
+
+```xml
+<xsl:stylesheet version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
+    <xsl:output method="xml" omit-xml-declaration="yes"/>
+    <xsl:variable name="logContent" select="/UpdateLogRequest/LogContent"/>
+    <xsl:variable name="transactionId" select="$logContent/UserContext/Context/Row/Common/TransactionId"/>
+    <xsl:template match="/">
+        <Result>
+        <Dataset name='special-data'>
+            <Row>
+            <Records>
+                <Rec>
+                <transaction_id><xsl:value-of select="$transactionId"/></transaction_id>
+                <request_data>
+                    <xsl:text disable-output-escaping="yes">&amp;lt;![CDATA[COMPRESS('</xsl:text>
+                    <xsl:copy-of select="$logContent/UserContent/Context"/>
+                    <xsl:text disable-output-escaping="yes">')]]&amp;gt;</xsl:text>
+                </request_data>
+                <request_format>SPECIAL</request_format>
+                <type>23</type>
+                </Rec>
+            </Records>
+            </Row>
+        </Dataset>
+        </Result>
+    </xsl:template>
+</xsl:stylesheet>
+```
+
+**WsFoobar.ecm**
+
+    ESPrequest FoobarSearchRequest
+    {
+        int RequestOption;
+        string RequestName;
+        [optional("hidden")] bool HiddenOption;
+    };
+
+    ESPresponse FoobarSearchResponse
+    {
+        int FoundCount;
+        string FoundAddress;
+    };
+
+    ESPservice [
+        auth_feature("DEFERRED"),
+        version("1"),
+        default_client_version("1"),
+    ] WsFoobar
+    {
+        ESPmethod FoobarSearch(FoobarSearchRequest, FoobarSearchResponse);
+    };
+
+#### Bundle
+
+The bundle includes all the components of the service binding in addition to the ESDL definition.
```

Comment:
Consider rephrasing. Between this and 426, seems somewhat circular.

Reply:
Removed that wording and instead called back to the usage for binding/bundle around line 45.